.github/workflows/sync_typeshed.yml

```diff
@@ -0,0 +1,32 @@
+name: Sync typeshed
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 1,15 * *"
+
+permissions:
+  contents: write
```

Comment:
Is this necessary? Apparently it gives the permissions in https://docs.github.com/en/rest/overview/permissions-required-for-github-apps#contents but on a quick glance none seem relevant.

Reply:
We push to a branch on the repo, so we do need some sort of write perm. Not sure your link corresponds exactly to this, since pushing isn't part of the REST API.

Reply:
I do think we could get rid of `issues: write` from here, because unlike stubsabot this workflow doesn't currently create an issue on failure

Reply:
I couldn't find documentation directly on what `contents: write` means, and somebody on StackOverflow said the permission model is the same as the REST API. But it does make sense that pushing is enabled by this permission.